internal/controller/datadogagentinternal/controller_reconcile_v2.go

```diff
@@ -64,19 +65,22 @@ func (r *Reconciler) reconcileInstanceV2(ctx context.Context, logger logr.Logger
 	r.updateMetricsForwardersFeatures(instanceCopy, enabledFeatures)
 
 	// 1. Manage dependencies.
-	depsStore, resourceManagers := r.setupDependencies(instanceCopy, logger)
+	// set the original DDAI as the owner of dependencies
+	depsStore, resourceManagers := r.setupDependencies(instance, logger)
 
 	var err error
-	if err = r.manageGlobalDependencies(logger, instanceCopy, resourceManagers, requiredComponents); err != nil {
-		return r.updateStatusIfNeededV2(logger, instance, newStatus, reconcile.Result{}, err, now)
-	}
-	if err = r.manageFeatureDependencies(logger, enabledFeatures, resourceManagers); err != nil {
-		return r.updateStatusIfNeededV2(logger, instance, newStatus, reconcile.Result{}, err, now)
-	}
-	if err = r.overrideDependencies(logger, resourceManagers, instanceCopy); err != nil {
-		return r.updateStatusIfNeededV2(logger, instance, newStatus, reconcile.Result{}, err, now)
+	// only manage dependencies for default DDAIs
+	if instance.Labels[agentprofile.ProfileLabelKey] == "" {
```

Comment:
```suggestion
	if !isDDAILabeledWithProfile(instance) {
```
Let's use the helper that is "safe" (check for nil labels -> that should not happen as we always inject the DDA name with `generateObjMetaFromDDA` but since the function already exists, might as well use it)